generated/nidaqmx/_library_interpreter.py

```diff
@@ -2232,11 +2232,52 @@ def get_ai_chan_cal_exp_date(self, task, channel_name):
 
     def get_analog_power_up_states(
             self, device_name, channel_name, channel_type):
-        raise NotImplementedError
+        state = []
+
+        args = [device_name]
+        argtypes = [ctypes_byte_str]
+
+        for index in range(len(channel_name)):
+            state_element = ctypes.c_double
+            state.append(state_element)
+
+            args.append(channel_name[index])
+            argtypes.append(ctypes_byte_str)
+            
+            args.append(ctypes.byref(state_element))
+            argtypes.append(ctypes.POINTER(ctypes.c_double))
+            
+            args.append(channel_type[index])
+            argtypes.append(ctypes.c_int32)
+            
+        args.append(None)
+
+        cfunc = lib_importer.cdll.DAQmxGetAnalogPowerUpStates
+        with cfunc.arglock:
+            cfunc.argtypes = argtypes
+            error_code = cfunc(*args)
+        check_for_error(error_code)
+        return state
```

Comment:
Ditto here, this isn't right... this needs to be a constructed list of objects

Reply:
I have updated the code to construct lists before returning.